### PR TITLE
test: add comprehensive unit tests for controllerUtils

### PR DESCRIPTION
**Pull request by muhammadrashid4587**
## Summary
- Adds unit tests for all 10 pure utility functions in `controllerUtils.ts`
- Covers: `requireString`, `optionalString`, `optionalNumber`, `optionalBoolean`, `parseSortOrder`, `requireTeamId`, `requireUserId`, `requireUserEmail`, `requireFirstName`, `requireUserRoles`
- Tests cover valid inputs, boundary values (empty strings, NaN, Infinity, whitespace), type coercion (string-to-number, string-to-boolean), and error conditions

## Motivation
`controllerUtils.ts` contains validation functions used across every controller in the server, but had zero test coverage. These pure functions are ideal unit test candidates.

## Test Plan
- [ ] Run `cd server && npm test` to verify all tests pass
- [ ] Verify no existing tests are broken

### CODE REVIEW

File: server/test/controllerUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
+import { jest } from "@jest/globals";
+import {
+	requireString,
+	optionalString,
+	optionalNumber,
+	optionalBoolean,
+	parseMonitorTypeFilter,
```
Comment (ajhollid) [inline server/test/controllerUtils.test.ts:7]: This is imported but not tested

File: server/test/controllerUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	it("throws when first name is empty string", () => {
+		expect(() => requireFirstName("")).toThrow("First name is required");
+	});
+});
+
+describe("requireUserRoles", () => {
+	it("returns the roles when valid", () => {
+		const roles = ["admin" as any];
+		expect(requireUserRoles(roles)).toEqual(roles);
+	});
```
Comment (ajhollid) [inline server/test/controllerUtils.test.ts:218]: Casting to `any` defeats the purpose here.  Essentially this says

```
const a = 1;
expect a = 1;
```

Yeah.. its true, but what's the point?

File: server/test/controllerUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		expect(() => optionalNumber("abc", "count")).toThrow("count must be a number");
+	});
+
+	it("throws for empty string", () => {
+		expect(() => optionalNumber("", "count")).toThrow("count must be a number");
+	});
+
+	it("throws for boolean", () => {
+		expect(() => optionalNumber(true, "count")).toThrow("count must be a number");
+	});
```
Comment (ajhollid) [inline server/test/controllerUtils.test.ts:103]: What about `null`?  Let's add a test case for that.

**Review by ajhollid — CHANGES_REQUESTED** [review]
Decent set of tests for the most part.  Some tightening up is needed and we're good to go here.  Thanks!

**Comment by Br0wnHammer** [comment]
Closing as stale.
